az-digital/az_quickstart
# Fixes #2247 Publication BibTeX Import

## Description
This PR adds an importer that creates Publication content from [BibTeX](https://www.bibtex.com/g/bibtex-format/) documents. To support this, a migration data parser `az_bibtex` (via providing a wrapper around `renanbr/bibtex-parser`) is added handles the parsiing of BibTex documents into migration source rows.

As many services that track publications allow for BibTeX export, this serves as a path for importing Publication content from other sites.

## Related issues
#2247 
#2237 (part of, does not close)

## How to test
- enable `az_publication` and `az_publication_bibtex` and `az_publication_doi`
- visit `/admin/content/az-publication/import/bibtex`
- upload a BibTex document and press `import`
- visit `/admin/content/az-publication/import/doi`
- enter a valid DOI and press `import`
- verify that for each import appropriate links are displayed via status message to the content via the title of the publication imported
- create a publication view
- inspect imported output

## Types of changes


### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

### Drupal core
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

### Drupal contrib projects
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

## Checklist
